lib/cusparse/helpers.jl

```diff
@@ -1,9 +1,26 @@
-## sparse matrix descriptor
+## dense vector descriptor
 
-mutable struct CuSparseMatrixDescriptor
+mutable struct CuDenseVectorDescriptor
+    handle::cusparseDnVecDescr_t
+
+    function CuDenseVectorDescriptor(v::CuVector{T}) where {T}
+        vec_ref = Ref{cusparseDnVecDescr_t}()
+        cusparseCreateDnVec(vec_ref, length(v), v, T)
+        obj = new(vec_ref[])
+        finalizer(cusparseDestroyDnVec, obj)
+        obj
+    end
+end
+
+Base.unsafe_convert(::Type{cusparseDnVecDescr_t}, desc::CuDenseVectorDescriptor) = desc.handle
+
+
+## matrix descriptor
+
+mutable struct CuMatrixDescriptor
     handle::cusparseMatDescr_t
 
-    function CuSparseMatrixDescriptor()
+    function CuMatrixDescriptor()
```

Comment:
Note the name change from CuSparseMatrixDescriptor to match the CUDA API `cusparseMatDescr_t` and `cusparseSpMatDescr_t`.